tests/feedback.test.ts

```diff
@@ -0,0 +1,147 @@
+import { validate } from 'class-validator';
+import { plainToClass } from 'class-transformer';
+import { DatabaseConnection, UserFactory } from './data';
+import { FeedbackFactory } from './data/FeedbackFactory';
+import { FeedbackControllerMock } from './controllers';
+import { ActivityScope, ActivityType, FeedbackStatus, UserAccessType } from '../types';
+import { ActivityModel } from '../models/ActivityModel';
+import { FeedbackModel } from '../models/FeedbackModel';
+import { Feedback } from '../api/validators/FeedbackControllerRequests';
+import { UserModel } from '../models/UserModel';
+import { Config } from '../config';
+
+beforeAll(async () => {
+  await DatabaseConnection.connect();
+});
+
+beforeEach(async () => {
+  await DatabaseConnection.clear();
+});
+
+afterAll(async () => {
+  await DatabaseConnection.clear();
+  await DatabaseConnection.close();
+});
+
+describe('feedback submission', () => {
+  test('returns proper feedback response on successful submission', async () => {
+    const [user] = UserFactory.create(1);
+    const [feedback] = FeedbackFactory.create(1);
+
+    await FeedbackControllerMock.mock([user]);
+    const submittedFeedback = await FeedbackControllerMock.submitFeedback(feedback, user);
+
+    expect(submittedFeedback.user).toStrictEqual(user.getPublicProfile());
+    expect(submittedFeedback.status).toStrictEqual(FeedbackStatus.SUBMITTED);
+    expect(submittedFeedback).toMatchObject(feedback);
+  });
+
+  test('is invalidated when submission description is too short', async () => {
+    const [feedback] = FeedbackFactory.with({ description: 'A short description' });
+
+    const errors = await validate(plainToClass(Feedback, feedback));
+
+    expect(errors).toBeDefined();
+    expect(errors).toHaveLength(1);
+    expect(errors[0].property).toEqual('description');
+    expect(errors[0].constraints.minLength).toBeDefined();
+  });
+
+  test('persists submission data and activity in database on successful submission', async () => {
```

Comment:
I think it'd be a little cleaner here if we used the GET route instead of querying the database, plus we get to make sure that route's working too.

Reply:
Wouldn't that make this test case dependent on whether or not a different controller is working? If that controller wasn't working, then this test case would fail, which might cause some confusion / extra looking since the actual feedback functionality would be working

Reply:
The GET's a pretty simple route, just a permissions check and db query.